### PR TITLE
incremental: wire up the events system in new backend

### PR DESCRIPTION
**Pull request by benjdlambert**
- chore: wire up subscriptions for new frontend system
- chore: wire up new events Signed-off-by: blam <ben@blam.sh>

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend-module-incremental-ingestion/src/module/WrapperProviders.ts [hunk 1]
```diff
@@ -130,6 +132,18 @@ export class WrapperProviders {
...
       });
+
+      const topics = engine.supportsEventTopics();
+      if (topics.length > 0) {
+        logger.info(
+          `Provider ${provider.getProviderName()} subscribing to events for topics: ${topics.join()}`,
+        );
+        await this.options.events.subscribe({
+          topics,
+          id: provider.getProviderName(),
```
Comment (freben) [inline plugins/catalog-backend-module-incremental-ingestion/src/module/WrapperProviders.ts:143]: is this id unique enough or will it need a prefix like `catalog-backend-module-incremental-ingestion:` before it or so

**Review by freben — APPROVED** [review]

File: plugins/catalog-backend-module-incremental-ingestion/src/module/WrapperProviders.ts [hunk 2]
```diff
@@ -130,6 +132,18 @@ export class WrapperProviders {
         frequency,
         timeout: length,
       });
+
+      const topics = engine.supportsEventTopics();
+      if (topics.length > 0) {
+        logger.info(
+          `Provider ${provider.getProviderName()} subscribing to events for topics: ${topics.join()}`,
```
Comment (freben) [inline plugins/catalog-backend-module-incremental-ingestion/src/module/WrapperProviders.ts:139]:
```suggestion
          `Provider ${provider.getProviderName()} subscribing to events for topics: ${topics.join(', ')}`,
```

**Review by freben — COMMENTED** [review]
one more

File: plugins/catalog-backend-module-incremental-ingestion/src/module/WrapperProviders.ts (hunk 1 above)
Comment (benjdlambert) [reply plugins/catalog-backend-module-incremental-ingestion/src/module/WrapperProviders.ts:143]: I don't know to be honest - it just says in the tsdoc that it's used for logging purposes.

Comment (Rugvip) [reply plugins/catalog-backend-module-incremental-ingestion/src/module/WrapperProviders.ts:143]: https://github.com/backstage/backstage/pull/27766 😉

**Pull request merged**